network/nattype.go

```diff
@@ -0,0 +1,26 @@
+package network
+
+// NATDeviceType indicates the type of the NAT device i.e. whether it is a Hard or an Easy NAT.
+type NATDeviceType int
```

Comment:
Is it possible to have this definition of nat type be merged with the existing [network.Reachability](https://github.com/libp2p/go-libp2p-core/blob/3267df19099a1ba68aa32103b13e925efeb2ef04/network/network.go#L74)

this seems to be a sub-division of what's currently used for network.ReachabilityPrivate.

Reply:
@willscott 

Considering that we need this event on a per transport protocol basis, it'll be extremely tricky to combine this with the `Reachability` event. I've updated the docs to inform the users that they should ALSO consume the Reachability event and use this NAT type event ONLY if Reachability=Private.